desktop/cmp/grid/GridModel.js

```diff
@@ -331,6 +331,15 @@ export class GridModel {
         this.columns = newCols;
     }
 
+    /**
+     * Grouped columns are a tree structure, this method returns a flat array representing the leaves
+     */
+
+    getColumnArray() {
+        const columns = this.cloneColumns();
+        return this.gatherLeaves(columns);
+    }
```

Comment:
This business used to live in grid state plug in, but as the colChooser and export plug in need this as well it makes sense to make it a public method on gridModels.

Reply:
Can we rename this getLeafColumns().  Seems much clearer and we would not need the comment.  Also, you don't *always* need to clone the result of this (in one case above you are just walking through it).  So better to just clone the result of this function when you need to.